simtools/visualization/legend_handlers.py

```diff
@@ -200,67 +222,86 @@ def legend_artist(legend, orig_handle, fontsize, handlebox):
     """
     Legend handler class to plot a representation of an LST in an array layout.
     """
 
-    @staticmethod
-    def legend_artist(legend, orig_handle, fontsize, handlebox):
+    def legend_artist(self, legend, orig_handle, fontsize, handlebox):
         center = (
             handlebox.xdescent + 0.3 * handlebox.width,
             handlebox.ydescent + 0.5 * handlebox.height,
         )
         radius = handlebox.height
         patch = mpatches.Circle(
             xy=center,
-            radius=radius,
+            radius=radius * self.radius_dict[lst] / self.radius_dict[lst],
             facecolor="none",
-            edgecolor="darkorange",
+            edgecolor=self.colors_dict[lst],
             transform=handlebox.get_transform(),
         )
         handlebox.add_artist(patch)
         return patch
 
 
-class MSTHandler(object):
+class MSTHandler(TelescopeHandler):
     """
     Legend handler class to plot a representation of an MST in an array layout.
     """
 
-    @staticmethod
-    def legend_artist(legend, orig_handle, fontsize, handlebox):
-        x0, y0 = handlebox.xdescent + 0.1 * handlebox.width, handlebox.ydescent
-        width = height = handlebox.height
-        patch = mpatches.Rectangle(
-            [x0, y0],
-            width,
-            height,
-            facecolor="dodgerblue",
-            edgecolor="dodgerblue",
+    def legend_artist(self, legend, orig_handle, fontsize, handlebox):
+        center = (
+            handlebox.xdescent + 0.25 * handlebox.width,
+            handlebox.ydescent + 0.5 * handlebox.height,
+        )
+        radius = handlebox.height
+        patch = mpatches.Circle(
+            xy=center,
+            radius=radius * self.radius_dict[mst] / self.radius_dict[lst],
+            facecolor=self.colors_dict[mst],
+            edgecolor=self.colors_dict[mst],
             transform=handlebox.get_transform(),
         )
         handlebox.add_artist(patch)
         return patch
 
 
-class SSTHandler(object):
+class SSTHandler(TelescopeHandler):
     """
     Legend handler class to plot a representation of an SST in an array layout.
     """
 
-    @staticmethod
-    def legend_artist(legend, orig_handle, fontsize, handlebox):
+    def legend_artist(self, legend, orig_handle, fontsize, handlebox):
         center = (
             handlebox.xdescent + 0.25 * handlebox.width,
             handlebox.ydescent + 0.5 * handlebox.height,
         )
         radius = handlebox.height
         patch = mpatches.Circle(
             xy=center,
-            radius=radius * (2.8 / 12),
-            facecolor="black",
-            edgecolor="black",
+            radius=radius * self.radius_dict[sst] / self.radius_dict[lst],
+            facecolor=self.colors_dict[sst],
+            edgecolor=self.colors_dict[sst],
+            transform=handlebox.get_transform(),
+        )
+        handlebox.add_artist(patch)
+        return patch
+
+
+class SCTHandler(object):
```

Comment:
I know that plotting is always messy, but having to duplicated the NNNHandler for telescope type is really very weird. I guess you've discussed this with Orel and there is no better solution?

Reply:
Yes, this is something Orel did. I think we do have to have it like this, since the legends are different for the different type of telescopes.

Reply:
Yes, there's no choice, that's how matplotlib handles legend handlers.